hashicorp/boundary-ui
# Icu 12860 retain values in form fields in credential library when switching types

:tickets: [Jira ticket](https://hashicorp.atlassian.net/browse/ICU-12860)

## Description
- Extract the name and description properties from the object returned by the ternary operator. If this.currentModel exists and is new, it will extract these properties from this.currentModel. Otherwise, it will try to extract them from an empty object, which will result in name and description being undefined.
- Fixed order of attributes to align with code base standard


## Screenshots (if appropriate):

https://github.com/hashicorp/boundary-ui/assets/24277002/c9aaebdd-9348-4b9c-83da-217b5634a62c


## How to Test
- Navigate to Orgs and any Project
- Navigate to Credential Stores
- Click on any Vault Store in list
- Click on the Credential Libraries tab
- Click on New Credential Library from manage dropdown
- Add a `name` and `description` and select then select the unselected type card and ensure that the data in those two fields persist





:technologist: [Admin preview](PATH_TO_FEATURE)

:desktop_computer: [Desktop preview](PATH_TO_FEATURE)

## Checklist:


~~- [ ] I have added JSON response output for API changes~~
~~- [ ] I have commented on my code, particularly in hard-to-understand areas~~
~~- [ ] I have added tests that prove my fix is effective or that my feature works~~
